module/documents/actor/actor.mjs

```diff
@@ -103,13 +102,9 @@ export default class Actor5e extends Actor {
 
   /** @inheritDoc */
   prepareBaseData() {
-
-    // Delegate preparation to type-subclass
-    if ( this.type === "group" ) {  // Eventually other types will also support this
-      return this.system._prepareBaseData();
-    }
-
-    this._prepareBaseArmorClass();
+    if ( !game.template.Actor.types.includes(this.type) ) return;
+    if ( this.type !== "group" ) this._prepareBaseArmorClass();
+    else if ( game.release.generation < 11 ) this.system.prepareBaseData();
```

Comment:
I checked the core API and `system.prepareBaseData()` will only be called automatically if `system` is an instance of `TypeDataModel`, which `GroupActor` is not anymore. So I think we must always call it manually until we drop v10 support and `SystemDataModel` can inherit from `TypeDataModel`.

```suggestion
    if ( this.type === "group" ) this.system.prepareBaseData();
    else this._prepareBaseArmorClass();
```

Reply:
Core checks if it is `TypeDataModel` or if it has a Function called `prepareBaseData`, so this is needed to avoid calling that method twice.

```javascript
// Core
if ( isTypeData || (this.system?.prepareBaseData instanceof Function) ) this.system.prepareBaseData();
```

Reply:
Ah, right, I totally glossed over that, that makes sense, thanks.